### PR TITLE
Add pagination sorting swagger

### PR DESCRIPTION
**Pull request by petro2061**
Added Pageable object to findAll method in Boot Controller class. Methods in the BookController class are also described, using the Open Api annotation.

### CODE REVIEW

File: src/main/java/project/onlinebookstore/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package project.onlinebookstore.config;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/config/OpenApiConfig.java]:
```suggestion
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -18,41 +22,71 @@
...
 
+    @Operation(summary = "Get all books",
+            description = "Allows get all books using pagination")
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(
+            @Parameter(description = "Standard object pageable "
+                    + "include page, size and sort")
+            Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java]:
```suggestion
            @ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 3]
```diff
@@ -18,41 +22,71 @@
...
+        return bookService.findAll(pageable);
     }
 
+    @Operation(summary = "Get book by Id",
+            description = "Allows get a book by Id")
     @GetMapping("/{id}")
-    public BookDto getBookById(@PathVariable Long id) {
+    public BookDto getBookById(
+            @Parameter(description = "Represents the book identifier")
+            @PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java]:
```suggestion
            @PathVariable @Positive Long id) {
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 4]
```diff
@@ -18,41 +22,71 @@
...
     }
 
+    @Operation(summary = "Delete book by Id",
+            description = "Allows delete a book by Id")
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
-    public void delete(@PathVariable Long id) {
+    public void delete(
+            @Parameter(description = "Represents the book identifier")
+            @PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java]:
```suggestion
            @PathVariable @Positive Long id) {
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 5]
```diff
@@ -18,41 +22,71 @@
...
     }
 
+    @Operation(summary = "Update book by Id",
+            description = "Allows update book by identifier")
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id,
-                              @RequestBody @Valid CreateBookRequestDto updateBookDto) {
+    public BookDto updateBookById(
+            @Parameter(description = "Represents the book identifier")
+            @PathVariable Long id,
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java]:
```suggestion
            @PathVariable @Positive Long id,
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 6]
```diff
@@ -18,41 +22,71 @@
...
     }
 
+    @Operation(summary = "Search book by parameters",
+            description = "Allows searching for books by certain parameters")
     @GetMapping("/search")
-    public List<BookDto> searchBooks(BookSearchParameters searchParameters) {
+    public List<BookDto> searchBooks(
+            @Parameter(description = "Represents an object whose fields represent"
+                    + " an array of search parameters")
+            BookSearchParameters searchParameters) {
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java]:
```suggestion
            BookSearchParameters searchParameters, @ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 7]
```diff
@@ -18,41 +22,71 @@
...
         return bookService.searchBooks(searchParameters);
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java]:
```suggestion
        return bookService.searchBooks(searchParameters, pageable);
```

File: src/main/java/project/onlinebookstore/controller/BookController.java [hunk 8]
```diff
@@ -18,41 +22,71 @@
 import project.onlinebookstore.dto.CreateBookRequestDto;
 import project.onlinebookstore.service.BookService;
 
-@RequiredArgsConstructor
+@Tag(name = "Book Management",
+        description = "Contains book entity management operations")
 @RestController
 @RequestMapping("/books")
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/controller/BookController.java:34]:
```suggestion
@RequestMapping("/books")
@Validated
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

**Comment by petro2061** [comment]
Fixed problem with codestyle and added new annotation to BookController class. In CreateBookRequestDto class changed annotation NotBlant to NotNull becouse i recived error: "No validator could be found for constraint 'jakarta.validation.constraints.NotBlank' validating type 'java.math.BigDecimal'. Check configuration for 'price'"

File: src/main/java/project/onlinebookstore/controller/BookController.java (hunk 5 above)
Comment (Elena-Bruyako) [reply src/main/java/project/onlinebookstore/controller/BookController.java]: not fixed

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see the comment

**Comment by petro2061** [comment]
Added annotation @Positive to field Long id.

**Review by Elena-Bruyako — APPROVED** [review]
Good job

**Pull request merged**